src/components/selection_input/selection_input.ts

```diff
@@ -198,6 +198,10 @@ export class SelectionInput extends Component<Props, SpreadsheetChildEnv> {
         updateSelectionWithArrowKeys(ev, this.env.model.selection);
       }
     } else if (ev.key === "Enter") {
+      if (!this.isConfirmable) {
+        return; // Return if the dataset is invalid
+      }
```

Comment:
Not enough to fix the problem, it can still occur if we call `props.onSelectionConfirmed` from the `removeInput` method.

We should either:
1) ensure that we never fire a `props.onSelectionConfirmed` in the selection input when a range is invalid
2) handle wrong ranges in the `onDataSeriesConfirmed` of the side panel

Not sure which one is best, I'd probably go with 2) because the configPanel is already aware of the invalid ranges via `onDataSeriesRangesChanged` so it make sense to handle them correctly on confirm.

Reply:
I have made the required changes and removed it from the `keydown` because it will now be handled in the `onDataSeriesConfirmed` function. Please review the changes and let me know if any potential issues.